### ISSUE TITLE
[Bug] Regression in parsing of (lower case) enum values between 6.22.0 and 6.23.0 of Microsoft.IdentityModel.JsonWebTokens

### ISSUE THREAD

**Issue by EraYaN** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#1941]
**Which version of Microsoft.IdentityModel are you using?**
Microsoft.IdentityModel.JsonWebTokens 6.23.0

**Where is the issue?**
 * [x] M.IM.JsonWebTokens
 * [ ] M.IM.KeyVaultExtensions
 * [ ] M.IM.Logging
 * [ ] M.IM.ManagedKeyVaultSecurityKey
 * [ ] M.IM.Protocols
 * [ ] M.IM.Protocols.OpenIdConnect
 * [ ] M.IM.Protocols.SignedHttpRequest
 * [ ] M.IM.Protocols.WsFederation
 * [ ] M.IM.TestExtensions
 * [ ] M.IM.Tokens
 * [ ] M.IM.Tokens.Saml
 * [ ] M.IM.Validators
 * [ ] M.IM.Xml
 * [ ] S.IM.Tokens.Jwt
 * Other (please describe)

**Is this a new or an existing app?**

The app is in production and I have upgraded to a new version of Microsoft.IdentityModel.

**Repro**

```csharp
enum TokenType {
	  Unknown
	  User,
	  Application
}

jsonWebToken.TryGetPayloadValue<TokenType>("type", out var tokenType )
```

Input token would have 

```json
"type": "user"
```

**Expected behavior**
It should support case insensitive parsing of enum values like with 6.22.0

**Actual behavior**
It returns the default value of the enum because it can't match the value anymore.

**Possible solution**

Maybe one of the `JsonSerializerOptions` `*CaseInsensitive` options.

**Additional context / logs / screenshots / links to code**
[Version diff](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/compare/6.22.0...6.23.0) seems like there was a switch to System.Text.Json for some of the claim handling, but enums were not considered.

**Comment by brentschmaltz** [comment]
@EraYaN i will look into this today.

**Comment by brentschmaltz** [comment]
@EraYaN this is definitely a regression. 
It doesn't appear to be about 'case', but that we ignore that the 'T' could be an enum [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/a502b0fc2d50ff7c9500062b098cfd056a81d283/src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs#L312).
I added the following code to the if statement starting on line 312 and the conversion to enum is successful.
I checked and the 6.22.0 release is case-insensitive.
```cs
if (typeof(T).IsEnum)
    return (T)Enum.Parse(typeof(T), jsonElement.ToString(), true);
```
I don't see an easy way for you to work around this issue, so we will release a fix.
There are some big changes from 6.22.0 - 6.23.1 so i would like to wait and see what other issues pop up before we release again.
We are seeing about a 25-30% perf improvement on validation with this release.

We should have this out within a week or so.
Sorry about the hassle.
